lib/providers/secret_storage.dart

```diff
@@ -0,0 +1,251 @@
+import 'dart:convert';
+import 'dart:math';
+
+import 'package:concordium_wallet/entities/password_hash.dart';
+import 'package:concordium_wallet/entities/secret_box.dart';
+import 'package:concordium_wallet/exceptions/secret_storage_exception.dart';
+import 'package:cryptography/cryptography.dart';
+import 'package:flutter/foundation.dart';
+import 'package:flutter_secure_storage/flutter_secure_storage.dart';
+import 'package:hive_flutter/hive_flutter.dart';
+
+abstract class SecretStorageProvider {
+  static String passwordObfuscationKey = "password_obfuscation";
+
+  /// Set an user password.
+  Future<void> setPassword(String password);
+
+  /// Checks if the users has set a password.
+  Future<bool> hasPassword();
+
+  /// From given user password validates if it is correct and unlocks
```

Comment:
I didn't catch at first when the method would return false (I had to check the test), so maybe we should write something along the lines of 
```suggestion
  /// Validates [password] and unlocks any encrypted storages if the password is correct. Otherwise it returns false.
```